src/storage/txn/commands.rs

```diff
@@ -317,6 +325,39 @@ macro_rules! tag {
     };
 }
 
+macro_rules! write_bytes {
+    ($field: ident) => {
+        fn write_bytes(&self) -> usize {
+            self.$field.as_encoded().len()
+        }
+    };
+    ($field: ident, multiple) => {
+        fn write_bytes(&self) -> usize {
+            self.$field.iter().map(|x| x.as_encoded().len()).sum()
+        }
+    };
+}
+
+macro_rules! gen_lock {
+    ($field: ident) => {
+        fn gen_lock(&self, latches: &Latches) -> latch::Lock {
+            latches.gen_lock(&[&self.$field])
+        }
+    };
+    ($field: ident: multiple) => {
+        fn gen_lock(&self, latches: &Latches) -> latch::Lock {
+            latches.gen_lock(&self.$field)
+        }
+    };
+    ($field: ident: multiple$transform: tt) => {
+        fn gen_lock(&self, latches: &Latches) -> latch::Lock {
+            #![allow(unused_parens)]
```

Comment:
Why not put parens in the macro? like
```
    ($field: ident: multiple($transform: tt)) => {
```

Reply:
> Why not put parens in the macro? like
> 
> ```
>     ($field: ident: multiple($transform: tt)) => {
> ```

It won't work, because 
> `tt` will match any single token or any pair of parenthesis/brackets/braces with their content.
And if the macro itself "consumes" the parens, `tt` won't get matched.